plugins/org.eclipse.elk.graph.text.ui/src/org/eclipse/elk/graph/text/ui/ConvertGraphHandler.xtend

```diff
@@ -0,0 +1,291 @@
+     * Transform the graph before it is written to the new file format.
+     */
+    protected def EObject transform(EObject model) {
+        val copy = EcoreUtil.copy(model)
+        if (targetExtension.equals(EXT_ELK_TEXT) && copy instanceof ElkNode) {
+            val graph = copy as ElkNode
+            // We want to convert to the textual format, so write missing identifiers into the graph
+            GraphIdentifierGenerator.forGraph(graph)
+                .assertValid
+                .assertExists
+                .assertUnique
+                .execute()
+            // Uncomment the following lines to reduce the graph to what's really relevant
+//             removeInvalidProperties(graph)
+//             removeUnnecessaryLayouts(graph)
+        }
+        return copy
+    }
+
+    /** 
+     * Create a target file URI from a source file.
+     */
+    private def createTarget(IFile file) {
+        val basePath = file.fullPath
+        val name = basePath.removeFileExtension.lastSegment
+        var IPath targetPath = new Path('''«name».«targetExtension»''')
+        var i = 0
+        val parent = file.parent
+        while (parent.exists(targetPath)) {
+            targetPath = new Path('''«name»«i++».«targetExtension»''')
+        }
+        targetPath = parent.fullPath.append(targetPath)
+        return URI.createPlatformResourceURI(targetPath.toString, false)
+    }
+
+    /** 
+     * Read the model contents from the given URI.
+     */
+    private def readModel(URI uri) {
+        val resourceSet = new ResourceSetImpl
+        val resource = resourceSet.getResource(uri, true)
+        return resource.contents
+    }
+
+    /** 
+     * Save a collection of models to the given URI.
+     */
+    private def saveModel(List<EObject> models, URI uri) throws IOException {
+        val resourceSet = new ResourceSetImpl
+        val resource = resourceSet.createResource(uri)
+        resource.contents.addAll(models)
+        resource.save(Collections.EMPTY_MAP)
+    }
+    
+    protected def removeInvalidProperties(ElkNode graph) {
+        val toRemove = newArrayList
+        graph.eAllContents.filter(EMapPropertyHolder).forEach[ element |
+            element.properties.forEach[ entry |
+                if (!isValid(entry.key, entry.value, element))
+                    toRemove += entry as ElkPropertyToValueMapEntryImpl
+            ]
+        ]
+        toRemove.forEach[EcoreUtil.remove(it)]
+    }
+    
+    static val EXCLUDED_OPTIONS = #[
+        CoreOptions.POSITION, CoreOptions.BEND_POINTS, CoreOptions.JUNCTION_POINTS
+    ]
+    
+    private def isValid(IProperty<?> property, Object value, EMapPropertyHolder container) {
+        // Check whether the property is a registered layout option
+        val option =
+            if (property instanceof LayoutOptionData)
+                property
+            else
+                LayoutMetaDataService.instance.getOptionData(property.id)
+        if (option === null)
```

Comment:
Personally, I'm not a big fan of leaving out the curly braces...

Reply:
I can understand that. Recently I learned that there are even [programming languages that enforce braces](https://golang.org/ref/spec#If_statements). But omitting the braces in Xtend code matches quite well the concise style of that language.

Reply:
That's interesting. I should look at Go after all... :D

Reply:
I don't like Go very much. It's certainly better than writing C, but IMHO the language design decisions are questionable.

I found this interesting:
https://medium.com/@george3d6/the-success-of-go-heralds-that-of-rust-73cb2e4c0500

Reply:
I'll take a look, thanks. :)